src/StatusQ/Controls/StatusAssetSelector.qml

```diff
@@ -10,149 +10,150 @@ import StatusQ.Components 0.1
 
 Item {
     id: root
+
     property var assets
     property var selectedAsset
     property string defaultToken: ""
     property string userSelectedToken: ""
     property var tokenAssetSourceFn: function (symbol) {
         return ""
     }
+
     // Define this in the usage to get balance in currency selected by user
     property var getCurrencyBalanceString: function (currencyBalance) { return "" }
-    implicitWidth: 106
-    implicitHeight: 32
 
     function resetInternal() {
         assets = null
         selectedAsset = null
     }
 
+    implicitWidth: 106
+    implicitHeight: comboBox.implicitHeight
+
     onSelectedAssetChanged: {
         if (selectedAsset && selectedAsset.symbol) {
-            iconImg.image.source = tokenAssetSourceFn(selectedAsset.symbol.toUpperCase())
-            selectedTextField.text = selectedAsset.symbol.toUpperCase()
+            d.iconSource = tokenAssetSourceFn(selectedAsset.symbol.toUpperCase())
+            d.text = selectedAsset.symbol.toUpperCase()
         }
     }
 
-    StatusSelect {
-        id: select
+    QtObject {
+        id: d
+        property string iconSource: ""
+        property string text: ""
+    }
+
+    StatusComboBox {
+        id: comboBox
+
         width: parent.width
-        bgColor: "transparent"
-        bgColorHover: Theme.palette.directColor8
+        height: parent.height
+
+        control.padding: 4
+        control.popup.width: 342
+        control.popup.x: width - control.popup.width
+
         model: root.assets
-        caretRightMargin: 0
-        select.radius: 6
-        select.height: root.height
-        selectMenu.width: 342
-        selectedItemComponent: Item {
-            anchors.fill: parent
+
+        control.background: Rectangle {
+            color: comboBox.control.hovered ? Theme.palette.directColor8 : "transparent"
+            radius: 6
+        }
+
+        contentItem: RowLayout {
+            spacing: 4
+
             StatusRoundedImage {
-                id: iconImg
-                anchors.left: parent.left
-                anchors.leftMargin: 4
-                width: 24
-                height: 24
-                anchors.verticalCenter: parent.verticalCenter
+                Layout.preferredWidth: 24
+                Layout.preferredHeight: 24
+                image.source: d.iconSource
```

Comment:
I think you still need verticalCenter alignment here (and for the text below too)